OpenJ9 sanity tests fail on OpenJ9 build with OpenJDK10 (18.3)

When running the Sanity tests against a build of OpenJ9 with  OpenJDK10 (18..3) the following test suites fail.
```
FAILED test targets:
	JCL_Test_SE90_0
	JCL_Test_SE90_1
```
These fail on these tests:
```
test_getProperties
test_getProperty
test_getProperty2
```
because the tests retrieve the system properties and check the java.version. 
The tests are looking for java.version starting with 9 where the java.version returned from this jdk is '10-internal'. 

Either the test frameworks needs to detect this new jdk version and run new tests or these tests need to be updated to work with both jdk9 and jdk10.

Exact test failure for one of the tests is:
```
test_getProperty2
java.lang.AssertionError: Failed to return correct property value: 10-internal
	at org.testng.AssertJUnit.fail(AssertJUnit.java:59) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.AssertJUnit.assertTrue(AssertJUnit.java:24) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.openj9.test.java.lang.Test_System.test_getProperty2(Test_System.java:220) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/Java8andUp/GeneralTest.jar)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@10-internal/Native Method) from jrt:/java.base
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@10-internal/NativeMethodAccessorImpl.java:62) from jrt:/java.base
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@10-internal/DelegatingMethodAccessorImpl.java:43) from jrt:/java.base
	at java.lang.reflect.Method.invoke(java.base@10-internal/Method.java:564) from jrt:/java.base
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestRunner.privateRun(TestRunner.java:756) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestRunner.run(TestRunner.java:610) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.runSuites(TestNG.java:1133) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.run(TestNG.java:1104) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.privateMain(TestNG.java:1434) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
	at org.testng.TestNG.main(TestNG.java:1403) from jdk.internal.loader.ClassLoaders$AppClassLoader@e62d135f(file:/root/openj9-openjdk-jdk9/openj9/jvmtest/TestConfig/lib/testng.jar)
```

--- a/test/Java8andUp/build.xml
+++ b/test/Java8andUp/build.xml
@@ -65,11 +65,10 @@
 		</if>
 		<!-- Compile the fake string impl and AppLoader caller with it -->
 		<if>
-			<equals arg1="${JAVA_VERSION}" arg2="SE90"/>
+			<equals arg1="${JAVA_VERSION}" arg2="SE80"/>
 			<then>
 				<javac srcdir="${NoSuchMethod_src}" destdir="${build}" includes="**/String.java,*AppLoaderCaller2.java" fork="true"
 				debug="on" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1">
-					<compilerarg line ="${compilerargline}" />
 					<classpath>
 						<pathelement location="../TestConfig/lib/asm-all.jar" />
 						<pathelement location="../TestConfig/lib/testng.jar" />
@@ -80,6 +79,7 @@
 			<else>
 				<javac srcdir="${NoSuchMethod_src}" destdir="${build}" includes="**/String.java,*AppLoaderCaller2.java" fork="true"
 				debug="on" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1">
+					<compilerarg line ="${compilerargline}" />
 					<classpath>
 						<pathelement location="../TestConfig/lib/asm-all.jar" />
 						<pathelement location="../TestConfig/lib/testng.jar" />
@@ -127,18 +127,31 @@
 		<echo>===debug:				on</echo>
 		<echo>===destdir:				${DEST}</echo>
 		<if>
-			<equals arg1="${JAVA_VERSION}" arg2="SE90"/>
+			<equals arg1="${JAVA_VERSION}" arg2="SE80"/>
 			<then>
+				<javac srcdir="${src}" destdir="${build}" debug="true" fork="true" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1">
+					<src path="${src}" />
+					<src path="${src_80}" />
+					<src path="${TestUtilities}" />
+					<src path="${transformerListener}" />
+					<exclude name="**/Cmvc194280.java" />
+					<exclude name="**/resources/**"/>
+					<exclude name="**/gpu/**"/>
+					<!-- requires special compilation methods -->
+					<exclude name="**/NoSuchMethod/**" />
+					<classpath>
+						<pathelement location="../TestConfig/lib/asm-all.jar" />
+						<pathelement location="../TestConfig/lib/testng.jar" />
+						<pathelement location="../TestConfig/lib/jcommander.jar" />
+						<pathelement location="${JAVA_BIN}/../../lib/tools.jar" />
+						<pathelement location="../TestConfig/lib/commons-cli.jar" />
+						<pathelement location="../TestConfig/lib/javassist.jar" />
+					</classpath>
+				</javac>
+			</then>
+			<else>
 				<property name="addModules" value="--add-modules java.se.ee,openj9.sharedclasses" />
-				<if>
-					<equals arg1="${JCL_VERSION}" arg2="latest"/>
-					<then>
-						<property name="addExports" value="--add-exports java.base/com.ibm.tools.attach.target=ALL-UNNAMED --add-exports java.base/com.ibm.oti.vm=ALL-UNNAMED --add-exports java.base/com.ibm.oti.util=ALL-UNNAMED --add-exports java.base/jdk.internal.vm.annotation=ALL-UNNAMED --add-exports java.base/jdk.internal.misc=ALL-UNNAMED --add-exports java.base/com.ibm.jit.crypto=ALL-UNNAMED --add-exports java.base/com.ibm.jit=ALL-UNNAMED --add-exports java.base/com.ibm.oti.reflect=ALL-UNNAMED --add-exports java.xml.bind/javax.xml.bind.annotation=ALL-UNNAMED --add-exports java.xml.bind/javax.xml.bind=ALL-UNNAMED" />
-					</then>
-					<else>
-						<property name="addExports" value="--add-exports java.base/com.ibm.tools.attach.target=ALL-UNNAMED --add-exports java.base/com.ibm.oti.vm=ALL-UNNAMED --add-exports java.base/com.ibm.oti.util=ALL-UNNAMED --add-exports java.base/jdk.internal.vm.annotation=ALL-UNNAMED --add-exports java.base/jdk.internal.misc=ALL-UNNAMED --add-exports java.base/com.ibm.jit.crypto=ALL-UNNAMED --add-exports java.base/com.ibm.jit=ALL-UNNAMED --add-exports java.base/com.ibm.oti.reflect=ALL-UNNAMED --add-exports java.xml.bind/javax.xml.bind.annotation=ALL-UNNAMED --add-exports java.xml.bind/javax.xml.bind=ALL-UNNAMED" />
-					</else>
-				</if>
+				<property name="addExports" value="--add-exports java.base/com.ibm.tools.attach.target=ALL-UNNAMED --add-exports java.base/com.ibm.oti.vm=ALL-UNNAMED --add-exports java.base/com.ibm.oti.util=ALL-UNNAMED --add-exports java.base/jdk.internal.vm.annotation=ALL-UNNAMED --add-exports java.base/jdk.internal.misc=ALL-UNNAMED --add-exports java.base/com.ibm.jit.crypto=ALL-UNNAMED --add-exports java.base/com.ibm.jit=ALL-UNNAMED --add-exports java.base/com.ibm.oti.reflect=ALL-UNNAMED --add-exports java.xml.bind/javax.xml.bind.annotation=ALL-UNNAMED --add-exports java.xml.bind/javax.xml.bind=ALL-UNNAMED" />
 				<javac srcdir="${src}" destdir="${build}" debug="true" fork="true" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1">
 					<src path="${src}" />
 					<src path="${src_90}" />
@@ -161,27 +174,6 @@
 						<pathelement location="../TestConfig/lib/javassist.jar" />
 					</classpath>
 				</javac>
-			</then>
-			<else>
-				<javac srcdir="${src}" destdir="${build}" debug="true" fork="true" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1">
-					<src path="${src}" />
-					<src path="${src_80}" />
-					<src path="${TestUtilities}" />
-					<src path="${transformerListener}" />
-					<exclude name="**/Cmvc194280.java" />
-					<exclude name="**/resources/**"/>
-					<exclude name="**/gpu/**"/>
-					<!-- requires special compilation methods -->
-					<exclude name="**/NoSuchMethod/**" />
-					<classpath>
-						<pathelement location="../TestConfig/lib/asm-all.jar" />
-						<pathelement location="../TestConfig/lib/testng.jar" />
-						<pathelement location="../TestConfig/lib/jcommander.jar" />
-						<pathelement location="${JAVA_BIN}/../../lib/tools.jar" />
-						<pathelement location="../TestConfig/lib/commons-cli.jar" />
-						<pathelement location="../TestConfig/lib/javassist.jar" />
-					</classpath>
-				</javac>
 			</else>
 		</if>
 	</target>
@@ -200,12 +192,12 @@
 	<!-- This target is used to prepare test resource which is located in org/openj9/resources-->
 	<target name="pack_resources" depends="compileNoSuchMethod" description="prepare JCL test resources">
 		<if>
-			<equals arg1="${JAVA_VERSION}" arg2="SE90"/>
+			<equals arg1="${JAVA_VERSION}" arg2="SE80"/>
 			<then>
-				<property name="addModules" value="--add-modules java.se.ee" />
+				<property name="addModules" value="" />
 			</then>
 			<else>
-				<property name="addModules" value="" />
+				<property name="addModules" value="--add-modules java.se.ee" />
 			</else>
 		</if>
 		<!-- compile java files within resources dir without '-parameter' flag in javac command line, README in '${src}/org/openj9/resources/methodparameters/README'-->

--- a/test/Java8andUp/playlist.xml
+++ b/test/Java8andUp/playlist.xml
@@ -679,7 +679,7 @@
 	</test>
 
 	<test>
-		<testCaseName>JCL_Test_SE90</testCaseName>
+		<testCaseName>JCL_Test_SE90+</testCaseName>
 		<variations>
 			<variation>NoOptions</variation>
 			<variation>-XX:RecreateClassfileOnload</variation>
@@ -728,6 +728,7 @@
 		</groups>
 		<subsets>
 			<subset>SE90</subset>
+			<subset>SE100</subset>
 		</subsets>
 	</test>
 

--- a/test/Java8andUp/src_90/org/openj9/test/java/lang/Test_System.java
+++ b/test/Java8andUp/src_90/org/openj9/test/java/lang/Test_System.java
@@ -144,10 +144,6 @@ public class Test_System {
 	 */
 	@Test
 	public void test_getProperties() {
-		Properties p = System.getProperties();
- 		AssertJUnit.assertTrue("Incorrect properties returned", p.getProperty(
- 				"java.version").indexOf("9", 0) == 0);
-
 		// ensure spec'ed properties are non-null. See System.getProperties()
 		// spec.
 		String[] props = { "java.version", "java.vendor", "java.vendor.url",
@@ -162,15 +158,6 @@ public class Test_System {
 		for (int i = 0; i < props.length; i++) {
 			AssertJUnit.assertTrue(props[i], System.getProperty(props[i]) != null);
 		}
-
-		/*
-		 * This test should only run for Java 1.8.0 and above. For Java 1.9.0
-		 * and above, we do not support java.ext.dirs property.
-		 */
-		if (p.getProperty("java.version").startsWith("1.8.0")) {
-			String javaExtDirs = "java.ext.dirs";
-			AssertJUnit.assertTrue(javaExtDirs, System.getProperty(javaExtDirs) != null);
-		}
 	}
 
 	/**
@@ -178,9 +165,6 @@ public class Test_System {
 	 */
 	@Test
 	public void test_getProperty() {
- 		AssertJUnit.assertTrue("Incorrect properties returned", System.getProperty(
- 				"java.version").indexOf("9", 0) == 0);
-
 		boolean is8859_1 = true;
 		String encoding = System.getProperty("file.encoding");
 		byte[] bytes = new byte[128];
@@ -217,10 +201,6 @@ public class Test_System {
 	 */
 	@Test
 	public void test_getProperty2() {
- 		AssertJUnit.assertTrue("Failed to return correct property value: "
- 				+ System.getProperty("java.version", "99999"), System
- 				.getProperty("java.version", "99999").indexOf("9", 0) >= 0);
-
 		AssertJUnit.assertTrue("Failed to return correct property value", System
 				.getProperty("bogus.prop", "bogus").equals("bogus"));
 	}

